lib/transport/selenium-webdriver/browserstackAutomate.js

```diff
@@ -0,0 +1,31 @@
+const BrowserStack = require('./browserstack.js');
+const AppiumMixin = require('./appiumMixin');
+const {Capabilities} = require('selenium-webdriver');
+
+class Automate extends BrowserStack {
+  get ApiUrl() {
+    return `https://api.browserstack.com/${this.productNamespace}`;
+  }
+
+  get productNamespace() {
+    return 'automate';
+  }
+
+  createSessionOptions(argv) {
+    this.extractAppiumOptions();
+
+    return super.createSessionOptions(argv) || this.desiredCapabilities;
+  }
+
+  createDriver({options}) {
+    if (options instanceof Capabilities) {
+      return super.createDriver({options});
+    }
+
+    return this.createAppiumDriver({options});
+  }
+}
```

Comment:
is this required. Can't we use the createDriver method from parent class which in this case would be selenium

Reply:
`createDriver` method from the parent class won't work in the cases when running tests on browsers not officially supported by Selenium (Samsung browser for ex.). In such cases, `options` passed to the `createDriver` method here would be `this.desiredCapabilities` and not `Options` class instance, due to which it will through error at later point when trying to access some methods of `Options` class which won't be available with `this.desiredCapabilities`.